src/Symfony/Component/Notifier/Bridge/Mastodon/Tests/MastodonTransportTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mastodon\Tests;
+
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\MockResponse;
+use Symfony\Component\Mime\Part\File;
+use Symfony\Component\Notifier\Bridge\Mastodon\MastodonOptions;
+use Symfony\Component\Notifier\Bridge\Mastodon\MastodonTransport;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Test\TransportTestCase;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Quentin Dequippe <quentin@dequippe.tech>
+ */
+class MastodonTransportTest extends TransportTestCase
+{
+    public function createTransport(HttpClientInterface $client = null): MastodonTransport
+    {
+        return (new MastodonTransport('testAccessToken', $client ?? $this->createMock(HttpClientInterface::class)))->setHost('host.test');
+    }
+
+    public function toStringProvider(): iterable
+    {
+        yield ['mastodon://host.test', $this->createTransport()];
+    }
+
+    public function supportedMessagesProvider(): iterable
+    {
+        yield [new ChatMessage('Hello World!')];
+    }
+
+    public function unsupportedMessagesProvider(): iterable
```

Comment:
did we move to static data providers already? can you rebase and figure out please?

Reply:
I rebased the PR, is it ok?
Do you have any example of static data providers?

Reply:
https://github.com/symfony/symfony/pull/48283

but it's not merged yet so let's forget for now

Reply:
(but the rebase might need to be replayed, see history, it's broken. Ensure you fetch+checkout  6.3 before rebase -i)

Reply:
I replayed a rebase, I don't know if it is ok now? Sorry I did not notice the branch switch 6.2 -> 6.3